src/lib/y2network/interfaces_collection.rb

```diff
@@ -87,5 +96,70 @@ def ==(other)
     end
 
     alias_method :eql?, :==
+
+    # Returns the interfaces that are enslaved in the given bridge
+    #
+    # @param master [String] bridge name
+    # @return [Array<String>] a list of interface names
+    # TODO: move to class of interface type br
+    def bridge_slaves(master)
+      bridge_index.select { |_k, v| v == master }.keys
+    end
+
+    # Creates list of devices enslaved in the bond device.
+    #
+    # @param bond_iface [String] a name of an interface of bond type
+    # @return list of names of interfaces enslaved in the bond_iface
+    # TODO: move to class of interface type bond
+    def bond_slaves(bond_iface)
+      bond_map = Yast::NetworkInterfaces::FilterDevices("netcard").fetch("bond", {}).fetch(bond_iface, {})
+
+      bond_map.select { |k, _| k.start_with?("BONDING_SLAVE") }.values
+    end
+
+    # Creates a map where the keys are the interfaces enslaved and the values
+    # are the bridges where them are taking part.
+    def bridge_index
+      index = {}
+
+      # TODO: should use by_type
+      bridge_devs = Yast::NetworkInterfaces.FilterDevices("netcard").fetch("br", {})
+
+      bridge_devs.each do |bridge_master, value|
+        value["BRIDGE_PORTS"].to_s.split.each do |if_name|
+          index[if_name] = bridge_master
+        end
+      end
+
+      index
+    end
+
+    def bond_index
+      index = {}
+
+      # TODO: should use by_type
+      bond_devs = Yast::NetworkInterfaces.FilterDevices("netcard").fetch("bond", {})
+
+      bond_devs.each do |bond_master, _value|
+        bond_slaves(bond_master).each do |slave|
+          index[slave] = bond_master
+        end
+      end
+
+      log.debug("bond slaves index: #{index}")
+
+      index
+    end
+
+    def all
```

Comment:
I know there is a `fixme` :) but I feel it is wrong. If you need so, you could have created a wrapper class adding those bits there (and keeping the `InterfacesCollection` as clean as possible).

Reply:
well I feel that it is wrong if interfaces collection works only with configured interfaces. On the other hand it makes sense to me to have possibility to list ```all``` or ```configured``` interfaces in the future .... this functionality is used already in old LanItems but under not so pretty names like ```find_configured```

Reply:
But it does not belong to InterfacesCollection, but to readers/writers.